### ISSUE TITLE
[Bug]: after update to Core 2024.11.0 Huawei integration no longer works

### ISSUE THREAD

**Issue by cvanlith** [wlcrs/huawei_solar#846]
### Describe the issue

The Huawei panel no longer shows data

### Describe your Huawei Solar Setup

Inverter Type: SUN2000-10KTL-M1
Inverter Firmware version: v100r001c00spc159
sDongle present: Yes
sDongle Type: sDongleA-05 (WiFi / Ethernet) 
sDongle Connectivitiy: WiFi
sDongle Firmware: v100r001c00spc130
Power meter present: no
Optimizers Present: Yes
Battery: None
Battery Firmware version: 
Huawei Solar integration version:  1.4.1

### How do you connect to the inverter?

Via the SDongle, wireless connection

### Upload your Diagnostics File

no download diagnostics button!

### Upload your relevant debug logs

```text
2024-11-08 09:21:45.437 INFO (MainThread) [huawei_solar.huawei_solar] Waiting for connection ...
2024-11-08 09:21:50.439 ERROR (MainThread) [huawei_solar.huawei_solar] Timeout while waiting for connection. Reconnecting...
Traceback (most recent call last):
File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
return await fut
^^^^^^^^^
File "/usr/local/lib/python3.12/asyncio/locks.py", line 212, in wait
await fut
asyncio.exceptions.CancelledError
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 157, in _communication_lock
await asyncio.wait_for(
File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
async with timeouts.timeout(timeout):
File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
raise TimeoutError from exc_val
TimeoutError
2024-11-08 09:21:50.448 ERROR (MainThread) [huawei_solar.huawei_solar] Aborting client creation due to error.
Traceback (most recent call last):
File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
return await fut
^^^^^^^^^
File "/usr/local/lib/python3.12/asyncio/locks.py", line 212, in wait
await fut
asyncio.exceptions.CancelledError
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 199, in create
await huawei_solar._initialize()
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 118, in _initialize
self.time_zone = (await self.get(rn.TIME_ZONE)).value
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 271, in get
return (await self.get_multiple([name], slave))[0]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 308, in get_multiple
response = await self._read_registers(
^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 444, in _read_registers
async with self._communication_lock():
File "/usr/local/lib/python3.12/contextlib.py", line 210, in __aenter__
return await anext(self.gen)
^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/huawei_solar/huawei_solar.py", line 157, in _communication_lock
await asyncio.wait_for(
File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
async with timeouts.timeout(timeout):
File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
raise TimeoutError from exc_val
TimeoutError
```

### Please confirm the following:

- [X] I'm running the latest release of Home Assistant.
- [X] I'm running the latest release of this integration.
- [X] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [X] I did upload the relevant debug logs (via 'Enable Debug Logging'-feature or by manually configuring HA logging)

**Comment by cvanlith** [comment]
In the past this happened before. Previously restarting the inverter or HA or both fixed the issue. This time no such luck.

**Comment by f54rog** [comment]
In my case going back to core 2024.10.0 solves the problem

**Comment by Marko-112** [comment]
I have the same issue where the integration stopped working after the Core 2024.11.0 update. Will there be a fix for this as well?

**Comment by Geroldin** [comment]
Good Morning,
i was able to add this Integration in the Alpha Version 1.5.0a this morning to my HomeAssistant running 2024.11.1. 
It seems to be running without issues at the moment.
HA System Details:
Core 2024.11.1
Supervisor 2024.11.2
Operating System 13.2
Frontend 20241106.2

Solarsystem (all Huawei):
EMMA A02 Software: SmartHEMS V100R024C00SPC100 (planning to Update to 101 soon)
SUN2000-8KTL-M1 Software: V100R001C00SPC165
Battery-Mgmt LUNA2000-10KW-C1
Battery: LUNA2000-7-S1
SCharger-22KT-S0 Software: V100R023C10SPC120


Now just hoping that the Huawei Charger will be supported through the Emma Integration as well.

**Comment by cvanlith** [comment]
After trying several core versions with no luck, I checked the WLAN0 connection and it was missing. Stupid, I should have checked that first, but was a while that I had problems with that.
Next re-establishing the wifi was not as easy as I thought. HA GUI could not connected to the inverter. I tried manual connection, using
ha network update wlan0 --wifi-mode infrastructure --ipv4-method auto --wifi-auth wpa-psk --wifi-ssid 'SUN2000-BTxxx' --wifi-psk 'xxx'. This also did not work: Error: Activating connection failed, check connection settings.
In a very old post searching on this error I found the suggestion that the signal strenght was too weak.
After moving the rpi4/ha closer to the inverter all started working again.

It now nows with the latest core 2024.11.1. I have no idea why suddenly after months of stable connections I had to place the rpi4/ha closer to the inverter.
